spec/models/provider_spec.rb

```diff
@@ -3,6 +3,35 @@
 require "rails_helper"
 
 describe Provider do
+  describe "scopes" do
+    describe ".with_active_hei" do
+      let(:active_hei_provider) { create(:provider, :hei) }
+      let(:inactive_hei_provider) { create(:provider, :hei) }
+      let(:unaccredited_hei_provider) { create(:provider, :hei, :unaccredited) }
+      let(:active_scitt_provider) { create(:provider, :scitt) }
+      let(:inactive_scitt_provider) { create(:provider, :scitt) }
+      let(:unaccredited_scitt_provider) { create(:provider, :scitt, :unaccredited) }
+
+      before do
+        active_hei_provider
+        inactive_hei_provider.discard
+        unaccredited_hei_provider
+        active_scitt_provider
+        inactive_scitt_provider.discard
+        unaccredited_scitt_provider
+      end
+
+      it "returns providers that are accredited and have a valid accreditation_id" do
+        expect(described_class.with_active_hei).to include(active_hei_provider)
+        expect(described_class.with_active_hei).not_to include(inactive_hei_provider)
+        expect(described_class.with_active_hei).not_to include(unaccredited_hei_provider)
+        expect(described_class.with_active_hei).not_to include(active_scitt_provider)
+        expect(described_class.with_active_hei).not_to include(inactive_scitt_provider)
+        expect(described_class.with_active_hei).not_to include(unaccredited_scitt_provider)
```

Comment:
```suggestion
        expect(described_class.with_active_hei).to contain_exactly(active_hei_provider)
```

Reply:
Fixed in https://github.com/DFE-Digital/register-trainee-teachers/pull/5191/commits/30dbaf8c4d947a180326f71a193d2e2902db014f